### ISSUE TITLE
Implement Intake's new entry_point system

### ISSUE THREAD

**Issue by jhamman** [NCAR/intake-esm#121]
- [x] Include a detailed description of the bug or suggestion
- [x] `conda list` of the conda environment you are using
- [x] Minimal, self-contained copy-pastable example that generates the issue if possible. 

### Description

Intake has recently introduced a new plugin system using entry_points. It seems Intake-esm is yet to implement this system and, as such, intake is throwing warnings (see below) on import. 

### What I Did

```
import intake

/srv/conda/envs/notebook/lib/python3.7/site-packages/intake/source/discovery.py:136: FutureWarning: The drivers ['esm_metadatastore'] do not specify entry_points and were only discovered via a package scan. This may break in a future release of intake. The packages should be updated.
  FutureWarning)
```

conda list
<details>
# packages in environment at /srv/conda/envs/notebook:
#
# Name                    Version                   Build  Channel
_libgcc_mutex             0.1                        main    defaults
absl-py                   0.7.1                    py37_0    conda-forge
adal                      1.2.2                      py_0    conda-forge
affine                    2.2.2                      py_0    conda-forge
aiohttp                   3.5.4            py37h14c3975_0    conda-forge
alembic                   1.0.11                     py_0    conda-forge
antlr-python-runtime      4.7.2                 py37_1000    conda-forge
appdirs                   1.4.3                      py_1    conda-forge
asciitree                 0.3.3                      py_2    conda-forge
asn1crypto                0.24.0                py37_1003    conda-forge
astor                     0.7.1                      py_0    conda-forge
async-timeout             3.0.1                   py_1000    conda-forge
async_generator           1.10                       py_0    conda-forge
attrdict                  2.0.1                      py_0    conda-forge
attrs                     19.1.0                     py_0    conda-forge
backcall                  0.1.0                      py_0    conda-forge
beautifulsoup4            4.8.0                    py37_0    conda-forge
bleach                    3.1.0                      py_0    conda-forge
blinker                   1.4                        py_1    conda-forge
blosc                     1.17.0               he1b5a44_0    conda-forge
bokeh                     1.3.4                    py37_0    conda-forge
boost-cpp                 1.70.0               h8e57a91_2    conda-forge
boto3                     1.9.203                    py_0    conda-forge
botocore                  1.12.203                   py_0    conda-forge
branca                    0.3.1                      py_0    conda-forge
bzip2                     1.0.8                h516909a_0    conda-forge
c-ares                    1.15.0            h516909a_1001    conda-forge
ca-certificates           2019.6.16            hecc5488_0    conda-forge
cachetools                2.1.0                      py_0    conda-forge
cairo                     1.16.0            hfb77d84_1002    conda-forge
cartopy                   0.17.0          py37hd759880_1006    conda-forge
certifi                   2019.6.16                py37_1    conda-forge
certipy                   0.1.3                      py_0    conda-forge
cf-units                  2.1.3            py37hc1659b7_0    conda-forge
cffi                      1.12.3           py37h8022711_0    conda-forge
cfitsio                   3.470                hb60a0a2_1    conda-forge
cftime                    1.0.3.4         py37hd352d35_1001    conda-forge
chardet                   3.0.4                 py37_1003    conda-forge
click                     7.0                        py_0    conda-forge
click-plugins             1.1.1                      py_0    conda-forge
cligj                     0.5.0                      py_0    conda-forge
climlab                   0.7.5            py37h7eb8c7e_0    conda-forge
cloudpickle               1.2.1                      py_0    conda-forge
colorcet                  2.0.1                      py_0    conda-forge
configobj                 5.0.6                      py_0    conda-forge
configurable-http-proxy   4.1.0                  node11_1    conda-forge
cryptography              2.7              py37h72c5cf5_0    conda-forge
curl                      7.65.3               hf8cf82a_0    conda-forge
cycler                    0.10.0                     py_1    conda-forge
cytoolz                   0.10.0           py37h516909a_0    conda-forge
dask                      2.2.0                      py_0    conda-forge
dask-core                 2.2.0                      py_0    conda-forge
dask-glm                  0.1.0                         0    conda-forge
dask-kubernetes           0.9.1                      py_0    conda-forge
dask-labextension         1.0.0                    pypi_0    pypi
dask-ml                   1.0.0                      py_0    conda-forge
datashader                0.7.0                      py_0    conda-forge
datashape                 0.5.4                      py_1    conda-forge
dbus                      1.13.6               he372182_0    conda-forge
decorator                 4.4.0                      py_0    conda-forge
defusedxml                0.5.0                      py_1    conda-forge
distributed               2.2.0                      py_1    conda-forge
docopt                    0.6.2                      py_1    conda-forge
docrep                    0.2.7                      py_0    conda-forge
docutils                  0.14                  py37_1001    conda-forge
entrypoints               0.3                   py37_1000    conda-forge
eofs                      1.4.0                      py_0    conda-forge
esmf                      7.1.0             h9a7cb89_1006    conda-forge
esmpy                     7.1.0            py37h24bf2e0_3    conda-forge
expat                     2.2.5             he1b5a44_1003    conda-forge
fasteners                 0.14.1                     py_3    conda-forge
fiona                     1.8.6            py37hf242f0b_3    conda-forge
fontconfig                2.13.1            h86ecdb6_1001    conda-forge
freetype                  2.10.0               he983fc9_0    conda-forge
freexl                    1.0.5             h14c3975_1002    conda-forge
fribidi                   1.0.5             h516909a_1002    conda-forge
fsspec                    0.4.1                      py_0    conda-forge
future                    0.17.1                py37_1000    conda-forge
gast                      0.2.2                      py_0    conda-forge
gcsfs                     0.3.0                      py_0    conda-forge
gdal                      2.4.1            py37h5f563d9_8    conda-forge
geopandas                 0.5.1                      py_0    conda-forge
geos                      3.7.2                he1b5a44_1    conda-forge
geotiff                   1.4.3             hb6868eb_1001    conda-forge
geoviews                  1.6.2                      py_0    conda-forge
geoviews-core             1.6.2                      py_0    conda-forge
gettext                   0.19.8.1          hc5be6a0_1002    conda-forge
giflib                    5.1.7                h516909a_1    conda-forge
glib                      2.58.3            h6f030ca_1002    conda-forge
google-auth               1.6.3                      py_0    conda-forge
google-auth-oauthlib      0.4.0                      py_0    conda-forge
graphite2                 1.3.13            hf484d3e_1000    conda-forge
graphviz                  2.40.1               h5933667_1    conda-forge
grpcio                    1.16.1           py37hf8bcb03_1    defaults
gst-plugins-base          1.14.5               h0935bb2_0    conda-forge
gstreamer                 1.14.5               h36ae1b5_0    conda-forge
gsw                       3.3.1            py37h516909a_0    conda-forge
h5netcdf                  0.7.4                      py_0    conda-forge
h5py                      2.9.0           nompi_py37h513d04c_1104    conda-forge
harfbuzz                  2.4.0                h9f30f68_2    conda-forge
hdf4                      4.2.13            h9a582f1_1002    conda-forge
hdf5                      1.10.5          nompi_h3c11f04_1100    conda-forge
heapdict                  1.0.0                 py37_1000    conda-forge
holoviews                 1.12.1                   pypi_0    pypi
hvplot                    0.4.0                      py_1    conda-forge
icu                       64.2                 he1b5a44_0    conda-forge
idna                      2.8                   py37_1000    conda-forge
imageio                   2.5.0                    py37_0    conda-forge
intake                    0.5.3                      py_0    conda-forge
intake-esm                2019.8.5                   py_0    conda-forge
intake-xarray             0.3.1                      py_0    conda-forge
ipykernel                 5.1.2            py37h5ca1d4c_0    conda-forge
ipyleaflet                0.11.1                   py37_1    conda-forge
ipython                   7.7.0            py37h5ca1d4c_0    conda-forge
ipython_genutils          0.2.0                      py_1    conda-forge
ipywidgets                7.5.1                      py_0    conda-forge
iris                      2.2.0                 py37_1003    conda-forge
jedi                      0.14.1                   py37_0    conda-forge
jinja2                    2.10.1                     py_0    conda-forge
jmespath                  0.9.4                      py_0    conda-forge
joblib                    0.13.2                     py_0    conda-forge
jpeg                      9c                h14c3975_1001    conda-forge
json-c                    0.13.1            h14c3975_1001    conda-forge
json5                     0.8.5                      py_0    conda-forge
jsonschema                3.0.2                    py37_0    conda-forge
jupyter                   1.0.0                      py_2    conda-forge
jupyter-server-proxy      1.1.0                      py_0    conda-forge
jupyter_client            5.3.1                      py_0    conda-forge
jupyter_console           6.0.0                      py_0    conda-forge
jupyter_core              4.4.0                      py_0    conda-forge
jupyterhub                1.0.0                    py37_0    conda-forge
jupyterlab                1.0.4                    py37_0    conda-forge
jupyterlab_server         1.0.0                      py_1    conda-forge
kealib                    1.4.10            h58c409b_1005    conda-forge
keras                     2.2.4                    py37_1    conda-forge
keras-applications        1.0.7                      py_1    conda-forge
keras-preprocessing       1.0.9                      py_1    conda-forge
kiwisolver                1.1.0            py37hc9558a2_0    conda-forge
krb5                      1.16.3            h05b26f9_1001    conda-forge
libblas                   3.8.0               11_openblas    conda-forge
libcblas                  3.8.0               11_openblas    conda-forge
libcurl                   7.65.3               hda55be3_0    conda-forge
libdap4                   3.20.2            hd48c02d_1000    conda-forge
libedit                   3.1.20170329      hf8c457e_1001    conda-forge
libffi                    3.2.1             he1b5a44_1006    conda-forge
libgcc-ng                 9.1.0                hdf63c60_0    defaults
libgdal                   2.4.1                heae24aa_8    conda-forge
libgfortran-ng            7.3.0                hdf63c60_0    defaults
libgpuarray               0.7.6             h14c3975_1003    conda-forge
libiconv                  1.15              h516909a_1005    conda-forge
libkml                    1.3.0             h4fcabce_1010    conda-forge
liblapack                 3.8.0               11_openblas    conda-forge
libnetcdf                 4.6.2             h056eaf5_1002    conda-forge
libopenblas               0.3.6                h6e990d7_6    conda-forge
libpng                    1.6.37               hed695b0_0    conda-forge
libpq                     11.4                 hd9ab2ff_3    conda-forge
libprotobuf               3.9.1                h8b12597_0    conda-forge
libsodium                 1.0.17               h516909a_0    conda-forge
libspatialindex           1.9.0                he1b5a44_1    conda-forge
libspatialite             4.3.0a            h79dc798_1030    conda-forge
libssh2                   1.8.2                h22169c7_2    conda-forge
libstdcxx-ng              9.1.0                hdf63c60_0    defaults
libtiff                   4.0.10            h57b8799_1003    conda-forge
libtool                   2.4.6             h14c3975_1002    conda-forge
libuuid                   2.32.1            h14c3975_1000    conda-forge
libxcb                    1.13              h14c3975_1002    conda-forge
libxml2                   2.9.9                hee79883_2    conda-forge
llvmlite                  0.29.0           py37hfd453ef_1    conda-forge
locket                    0.2.0                      py_2    conda-forge
lz4                       2.1.9            py37hd79334b_0    conda-forge
lz4-c                     1.8.3             he1b5a44_1001    conda-forge
mako                      1.1.0                      py_0    conda-forge
markdown                  3.1.1                      py_0    conda-forge
markupsafe                1.1.1            py37h14c3975_0    conda-forge
matplotlib                3.1.1                    py37_1    conda-forge
matplotlib-base           3.1.1            py37he7580a8_1    conda-forge
mechanicalsoup            0.11.0                     py_0    conda-forge
mercantile                1.1.2                      py_0    conda-forge
metpy                     0.10.2                   py37_0    conda-forge
mistune                   0.8.4           py37h14c3975_1000    conda-forge
mock                      3.0.5                    py37_0    conda-forge
monotonic                 1.5                        py_0    conda-forge
mpi                       1.0                       mpich    conda-forge
mpich                     3.2.1             hc99cbb1_1013    conda-forge
msgpack-numpy             0.4.4.3                    py_0    conda-forge
msgpack-python            0.6.1            py37h6bb024c_0    conda-forge
multidict                 4.5.2           py37h14c3975_1000    conda-forge
multipledispatch          0.6.0                      py_0    conda-forge
munch                     2.3.2                      py_0    conda-forge
nbconvert                 5.5.0                      py_0    conda-forge
nbformat                  4.4.0                      py_1    conda-forge
nbgitpuller               0.7.0                      py_0    conda-forge
ncurses                   6.1               hf484d3e_1002    conda-forge
netcdf-fortran            4.4.5             hfc18c51_1001    conda-forge
netcdf4                   1.5.1.2          py37h73a1b54_1    conda-forge
networkx                  2.3                        py_0    conda-forge
ninja                     1.9.0                h6bb024c_0    conda-forge
nodejs                    11.14.0              he1b5a44_1    conda-forge
notebook                  6.0.0                    py37_0    conda-forge
numba                     0.45.1           py37hb3f55d8_0    conda-forge
numcodecs                 0.6.3            py37hf484d3e_0    conda-forge
numpy                     1.17.0           py37h95a1406_0    conda-forge
oauthlib                  3.0.1                      py_0    conda-forge
olefile                   0.46                       py_0    conda-forge
openjpeg                  2.3.1                h58a6597_0    conda-forge
openssl                   1.1.1c               h516909a_0    conda-forge
owslib                    0.18.0                     py_0    conda-forge
packaging                 19.0                       py_0    conda-forge
pamela                    1.0.0                      py_0    conda-forge
pandas                    0.25.0           py37hb3f55d8_0    conda-forge
pandoc                    2.7.3                         0    conda-forge
pandocfilters             1.4.2                      py_1    conda-forge
panel                     0.6.0                         1    conda-forge
pango                     1.42.4               he7ab937_0    conda-forge
param                     1.9.1                      py_0    conda-forge
parso                     0.5.1                      py_0    conda-forge
partd                     1.0.0                      py_0    conda-forge
pcre                      8.41              hf484d3e_1003    conda-forge
pexpect                   4.7.0                    py37_0    conda-forge
pickleshare               0.7.5                 py37_1000    conda-forge
pillow                    6.1.0            py37h6b7be26_1    conda-forge
pint                      0.9                      py37_2    conda-forge
pip                       19.2.1                   py37_0    conda-forge
pixman                    0.38.0            h516909a_1003    conda-forge
pooch                     0.5.2                    py37_0    conda-forge
poppler                   0.67.0               ha967d66_7    conda-forge
poppler-data              0.4.9                         1    conda-forge
postgresql                11.4                 hc63931a_3    conda-forge
proj4                     5.2.0             he1b5a44_1004    conda-forge
prometheus_client         0.7.1                      py_0    conda-forge
prompt_toolkit            2.0.9                      py_0    conda-forge
protobuf                  3.9.1            py37he1b5a44_0    conda-forge
psutil                    5.6.3            py37h516909a_0    conda-forge
pthread-stubs             0.4               h14c3975_1001    conda-forge
ptyprocess                0.6.0                   py_1001    conda-forge
pyasn1                    0.4.6                      py_0    conda-forge
pyasn1-modules            0.2.6                      py_0    conda-forge
pycparser                 2.19                     py37_1    conda-forge
pyct                      0.4.6                      py_0    conda-forge
pyct-core                 0.4.6                      py_0    conda-forge
pycurl                    7.43.0.2         py37h16ce93b_1    conda-forge
pydap                     3.2.2                 py37_1000    conda-forge
pyepsg                    0.4.0                      py_0    conda-forge
pygments                  2.4.2                      py_0    conda-forge
pygpu                     0.7.6           py37h3010b51_1000    conda-forge
pyjwt                     1.7.1                      py_0    conda-forge
pykdtree                  1.3.1           py37h3010b51_1002    conda-forge
pyke                      1.1.1                 py37_1000    conda-forge
pyopenssl                 19.0.0                   py37_0    conda-forge
pyorbital                 1.5.0                      py_0    conda-forge
pyparsing                 2.4.2                      py_0    conda-forge
pyproj                    1.9.6           py37h516909a_1002    conda-forge
pyqt                      5.9.2            py37hcca6a23_2    conda-forge
pyresample                1.12.3           py37hf2d7682_0    conda-forge
pyrsistent                0.15.4           py37h516909a_0    conda-forge
pyshp                     2.1.0                      py_0    conda-forge
pysocks                   1.7.0                    py37_0    conda-forge
pyspectral                0.8.9                      py_0    conda-forge
pyspharm                  1.0.9           py37h6091dcd_1003    conda-forge
python                    3.7.3                h33d41f4_1    conda-forge
python-blosc              1.8.1            py37hf484d3e_0    conda-forge
python-dateutil           2.8.0                      py_0    conda-forge
python-editor             1.0.4                      py_0    conda-forge
python-geotiepoints       1.1.8            py37hd352d35_0    conda-forge
python-kubernetes         10.0.0                   py37_0    conda-forge
python-snappy             0.5.4            py37hee44bf9_0    conda-forge
pytorch-cpu               1.1.0            py37he1b5a44_0    conda-forge
pytz                      2019.2                     py_0    conda-forge
pyviz_comms               0.7.2                      py_0    conda-forge
pywavelets                1.0.3            py37hd352d35_1    conda-forge
pyyaml                    5.1.2            py37h516909a_0    conda-forge
pyzmq                     18.0.2           py37h1768529_2    conda-forge
qt                        5.9.7                h0c104cb_3    conda-forge
qtconsole                 4.5.2                      py_0    conda-forge
rasterio                  1.0.24           py37hdff7cfa_0    conda-forge
readline                  8.0                  hf8c457e_0    conda-forge
regionmask                0.4.0                      py_0    conda-forge
requests                  2.22.0                   py37_1    conda-forge
requests-oauthlib         1.2.0                      py_0    conda-forge
rio-cogeo                 1.1.0                      py_0    conda-forge
rsa                       3.4.2                      py_1    conda-forge
rtree                     0.8.3           py37h666c49c_1002    conda-forge
ruamel                    1.0                      py37_0    conda-forge
ruamel.yaml               0.16.1           py37h516909a_0    conda-forge
s3fs                      0.3.3                      py_0    conda-forge
s3transfer                0.2.1                    py37_0    conda-forge
satpy                     0.16.1             pyh326bf55_0    conda-forge
scikit-image              0.15.0           py37hb3f55d8_2    conda-forge
scikit-learn              0.21.3           py37hcdab131_0    conda-forge
scipy                     1.3.0            py37h921218d_1    conda-forge
send2trash                1.5.0                      py_0    conda-forge
setuptools                41.0.1                   py37_0    conda-forge
shapely                   1.6.4           py37hec07ddf_1006    conda-forge
simpervisor               0.3                        py_1    conda-forge
sip                       4.19.8          py37hf484d3e_1000    conda-forge
six                       1.12.0                py37_1000    conda-forge
snappy                    1.1.7             he1b5a44_1002    conda-forge
snuggs                    1.4.6                      py_0    conda-forge
sortedcontainers          2.1.0                      py_0    conda-forge
soupsieve                 1.9.2                    py37_0    conda-forge
sqlalchemy                1.3.6            py37h516909a_0    conda-forge
sqlite                    3.29.0               hcee41ef_0    conda-forge
supermercado              0.0.5                      py_0    conda-forge
tblib                     1.4.0                      py_0    conda-forge
tensorboard               1.13.1                   py37_0    conda-forge
tensorflow                1.13.1                   py37_0    conda-forge
tensorflow-estimator      1.13.0                     py_0    defaults
termcolor                 1.1.0                      py_2    conda-forge
terminado                 0.8.2                    py37_0    conda-forge
testpath                  0.3.1                    pypi_0    pypi
theano                    1.0.3            py37hfc679d8_1    conda-forge
tk                        8.6.9             hed695b0_1002    conda-forge
toolchain                 2.4.0                         0    defaults
toolchain_c_linux-64      2.4.0                         0    defaults
toolchain_cxx_linux-64    2.4.0                         0    defaults
toolz                     0.10.0                     py_0    conda-forge
tornado                   6.0.3            py37h516909a_0    conda-forge
tqdm                      4.32.2                     py_0    conda-forge
traitlets                 4.3.2                 py37_1000    conda-forge
traittypes                0.2.1                      py_1    conda-forge
trollimage                1.9.0                      py_0    conda-forge
trollsift                 0.3.2                      py_0    conda-forge
tzcode                    2019a             h516909a_1002    conda-forge
udunits2                  2.2.27.6          h4e0c4b3_1001    conda-forge
urllib3                   1.25.3                   py37_0    conda-forge
wcwidth                   0.1.7                      py_1    conda-forge
webencodings              0.5.1                      py_1    conda-forge
webob                     1.8.5                      py_0    conda-forge
websocket-client          0.56.0                   py37_0    conda-forge
werkzeug                  0.15.5                     py_0    conda-forge
wheel                     0.33.4                   py37_0    conda-forge
widgetsnbextension        3.5.1                    py37_0    conda-forge
windspharm                1.7.0                 py37_1000    conda-forge
xarray                    0.12.3                     py_0    conda-forge
xerces-c                  3.2.2             h8412b87_1004    conda-forge
xesmf                     0.2.0                    py37_0    conda-forge
xgcm                      0.2.0                      py_0    conda-forge
xorg-kbproto              1.0.7             h14c3975_1002    conda-forge
xorg-libice               1.0.10               h516909a_0    conda-forge
xorg-libsm                1.2.3             h84519dc_1000    conda-forge
xorg-libx11               1.6.8                h516909a_0    conda-forge
xorg-libxau               1.0.9                h14c3975_0    conda-forge
xorg-libxdmcp             1.1.3                h516909a_0    conda-forge
xorg-libxext              1.3.4                h516909a_0    conda-forge
xorg-libxpm               3.5.12            h14c3975_1002    conda-forge
xorg-libxrender           0.9.10            h516909a_1002    conda-forge
xorg-libxt                1.1.5             h516909a_1003    conda-forge
xorg-renderproto          0.11.1            h14c3975_1002    conda-forge
xorg-xextproto            7.3.0             h14c3975_1002    conda-forge
xorg-xproto               7.0.31            h14c3975_1007    conda-forge
xrft                      0.2.0                      py_0    conda-forge
xrviz                     0.1.0+83.ge39c38a          pypi_0    pypi
xz                        5.2.4             h14c3975_1001    conda-forge
yaml                      0.1.7             h14c3975_1001    conda-forge
yarl                      1.3.0           py37h14c3975_1000    conda-forge
zarr                      2.3.2                    py37_0    conda-forge
zeromq                    4.3.2                he1b5a44_2    conda-forge
zict                      1.0.0                      py_0    conda-forge
zlib                      1.2.11            h516909a_1005    conda-forge
zstd                      1.4.0                h3b9ef0a_0    conda-forge
</details>

**Comment by andersy005** [comment]
@jhamman, this issue should be fixed once #119 is merged.

By the way, if you have a moment can I ask you to look through #119 ( I could use some feedback :))

**Comment by andersy005** [comment]
Addressed in #119
